### ISSUE TITLE
Server restarting every 2 secs when running in PM2

### ISSUE THREAD

**Issue by Jelleeelco** [AShifter/harmony-span#7]
Hi Logan,

Not sure if this is due to my system, but I observed the following. When running the harmony-span server from command line, it works perfectly. But I wanted to make it more robust (so in case my Raspberry Pi reboots it will also reboot) and installed PM2 to manage the .js application. When I go into PM2 monit, it shows that the server is restarting every two seconds. Looking at the log, it shows the following:
![image](https://user-images.githubusercontent.com/67953250/86894193-091a4280-c103-11ea-8ebc-9ffb85ae1bc5.png)

Questions:
- Do I have to maybe change the config.json file?
- Is PM2 not suiteable for this .js to run automatically in the background on a Rapsberry Pi?
- Is there an alternative way to run the server in the background and booting up automatically when restarting the PI?

Many thanks!
Jelle..

**Comment by Jelleeelco** [comment]
And when adding the harmony-span.js to /etc/rc.local to run during startup, it also returns the error as per above. This is reproducable by executing:

su pi -c 'node /home/pi/harmony-span/harmony-span.js < /dev/null &'

Thanks
Jelle..

**Comment by AShifter** [comment]
Thanks again for opening this issue.
Try navigating directly to the folder where you have HarmonySpan located and try running it from there - I suspect that HarmonySpan is looking for the config file in whatever directory you start it in (so if you're in /home/pi for instance and you run HarmonySpan from a different directory, HarmonySpan might be looking in /home/pi for the config file which isn't correct.

**Comment by Jelleeelco** [comment]
Hi Logan,

I think you're right, but I'm not sure how to fix that. I thought I did run
it from the actual folder before I created the startup file in PM2. My
second attempt to have it startup using /etc/rc.local, my line added to
RC.local does contain the actual folder.

I know I can run it in a screen am just detach it, but I want it "idiot
proof" so that when I'm not around and the pi needs to reboot, it's all
still there.

Sidenote: amazing project and amazing functionality you build! Thank you so
much!

Cheers
Jelle..
